spec/controllers/provider_management/external_users_controller_spec.rb

```diff
@@ -0,0 +1,245 @@
+
+RSpec.describe ProviderManagement::ExternalUsersController, type: :controller do
+  let(:case_worker_manager)   { create(:case_worker, :provider_manager) }
+  let(:provider)      { create(:provider) }
+  let(:frozen_time)  { 6.months.ago }
+  let(:external_user)   do
+    Timecop.freeze(frozen_time) { create(:external_user, :admin, provider: provider) }
+  end
+
+  before { sign_in case_worker_manager.user }
+
+  describe "GET #show" do
+    before { get :show, params: { provider_id: provider, id: external_user } }
+
+    it "returns http success" do
+      expect(response).to be_successful
+    end
+
+    it 'assigns @provider and @external_user' do
+      expect(assigns(:provider)).to eq(provider)
+      expect(assigns(:external_user)).to eq(external_user)
+    end
+  end
+
+  describe "GET #index" do
+    before { get :index, params: { provider_id: provider } }
+
+    it "returns http success" do
+      expect(response).to be_successful
+    end
+
+    it 'assigns @provider' do
+      expect(assigns(:provider)).to eq(provider)
+    end
+
+    it 'assigns @external_users' do
+      expect(assigns(:external_users)).to include(external_user)
+    end
+  end
+
+  describe 'GET #find' do
+    before { get :find }
+
+    it 'returns http success' do
+      expect(response).to be_successful
+    end
+  end
+
+  describe 'POST #search' do
+    subject { response }
+
+    before { post :search, params: { external_user: { email: email } } }
+
+    context 'when the email is for a provider' do
+      let(:email) { external_user.email }
+
+      it { is_expected.to redirect_to(provider_management_provider_external_user_path(external_user.provider, external_user)) }
+    end
+
+    context 'when the email is for a non-provider' do
+      let(:email) { case_worker_manager.email }
+
+      it { is_expected.to redirect_to(provider_management_external_users_find_path) }
+    end
+
+    context 'when the email does not exist' do
+      let(:email) { 'vail.email@does.not.exist.com' }
+
+      it { is_expected.to redirect_to(provider_management_external_users_find_path) }
+    end
+  end
+
+  describe "GET #new" do
+    let(:external_user) do
+     a = ExternalUser.new(provider: provider)
+     a.build_user
+     a
+    end
+
+    before { get :new, params: { provider_id: provider } }
+
+    it "returns http success" do
+      expect(response).to be_successful
+    end
+
+    # NOTE: use json comparison because we are not interested in
+    #       whether the object is the same just that it creates a
+    #       new one and builds its user
+    it 'assigns @provider and @external_user' do
+      expect(assigns(:provider)).to eq(provider)
+      expect(assigns(:external_user).to_json).to eq(external_user.to_json)
+    end
+
+    it 'builds user for @external_user' do
+      expect(assigns(:external_user).user.to_json).to eq(external_user.user.to_json)
+    end
+
+    it 'renders the new template' do
+      expect(response).to render_template(:new)
+    end
+
```

Comment:
minor/non-blocking (copy an paste i imagine) - unnecessary linebreaks here and below

Reply:
oh, i have learnt so much. and my preferences changed. 🤣